### ISSUE TITLE
New build - EXPERIMENTAL] 20w29c: Decals Final Pass

### ISSUE THREAD

**Issue by tsunamayo** [tsunamayo/Starship-EVO#2681]
Final pass on the decals technology. Ship preview will now have decals.
A lot of long time issues and limitations have been fixed.

New Features:
- Decals compatible with ship preview.
- Decals applied only on their master entity.

Breaking Changes:
- Some decals on wedge surface might appears cut after this update, you will need to replace them.

Bugfixes:
#1641 Decals not visible in ship preview.
#2135 Decals texture looks weird on white surface.
#1640 #1485 Decals visible on different entity.
#1469 Decals on wedge viewed at angle appears cut.
#2679 Decals show up on bricks placed above.
#2435 Decals positioning on wedge surface can lead them to be cut out.
#1579 Decals holo preview cut out on slopes.

**Comment by tsunamayo** [comment]
Build is up! Thanks for all the feedback guy!

**Comment by ZachZent** [comment]
Glad stickers are fixed. They are a big appeal for the game. 

Hopefully we can get some new stickers soon. There are a few suggestions in here for some new stickers
https://github.com/tsunamayo/Starship-EVO/issues/2461

**Comment by BrickMacklin** [comment]
Unfortunately I have a few issues. 
1. The striped "caution tape" paint has now disappeared into a single color. In the below photo the yellow is supposed to be yellow and black. The red is supposed to be red and white.

![image](https://user-images.githubusercontent.com/37836252/87843073-5fb01b00-c87f-11ea-8950-acf449c4cdea.png)

2. When viewing parts of the ship sometimes the view of the blocks disappear or glitch out. You can see it in the gif below where the interior blocks turn blue
https://imgur.com/fa8rsNz

3. The orange triangular stickers on either side of the frigate are flickering in and out frequently. Gif below shows this.
https://imgur.com/kw2njvc

**Comment by ProPeach** [comment]
Just a few issues on my side, but overall the stickers are soooo much better.
GPU - RX580
I think this first one is similar to #3 that BrickMalin posted - Some stickers (notice the red on the red and white fighter near the reticle) flicker at medium distances
https://i.imgur.com/DoLia1k.mp4

Also while all stickers now show up on the ship preview (which is awesome!), some of them have no colour even if you replace and repaint them
![image](https://user-images.githubusercontent.com/27857711/87850155-f7e0eb00-c8e5-11ea-8c61-f30e26fd4ead.png)

**Comment by ProPeach** [comment]
I'm not sure if this in the scope of your fixes, but issue #1772 is still present that causes a flickering effect on the brick borders when stickers are used next to them.

On the subject of brick borders, do you still plan on reducing their visual impact a little bit? I know you said that you would try a long while ago, but I'm just wondering if that's something you'd still like to do considering how unpopular they are

**Comment by ProPeach** [comment]
Another one - issue 1 from #2679 is fixed for full bricks, but not for any brick thinner than 25cm
![image](https://user-images.githubusercontent.com/27857711/87850471-58712780-c8e8-11ea-90f2-abf293eaacb1.png)

**Comment by tsunamayo** [comment]
Yes @ProPeach your issues are the same as @BrickMacklin it seems, I will investigate and keep you posted. It seems a bit less game breaking, I think I will push that build tomorrow to DEFAULT. Thanks all!

**Comment by Kaiser-Indrasil** [comment]
I seem to have some problem with stickers placed on... ekhm... *clipped* bricks:
![oZ4XpXQV2M](https://user-images.githubusercontent.com/63756057/87857234-16160d80-c91d-11ea-899f-0700c4fcb336.gif)

It seems that stickers now appear only on the surface they were applied on. Which leads me to the second issue:

![Starship_EVO_eCpsil6p3c](https://user-images.githubusercontent.com/63756057/87857260-4493e880-c91d-11ea-9e35-832f6aac3093.png)
There used to be a lot of stickers applied to bricks hidden less than 25 centimeters below the surfaces they appeared on. Now they're gone. Looks like I'll have to find another way to apply them.

**Comment by tsunamayo** [comment]
Yeah you where using two bugs: children entity clipping and decals visible on a different entity, so there is nothing I can do for you on that one, sorry ;)

**Comment by tsunamayo** [comment]
@BrickMacklin both your issues are fixed for new update. Thanks!

**Comment by tsunamayo** [comment]
@ProPeach for the preview issue what is the test case exactly? Is it related to repainting them - or is there is a mismatch out of the box? thx

**Comment by ProPeach** [comment]
There was a mismatch with ships already loaded in to a world that I loaded. When I tried placing and painting new sticker on the ship to double check, those were mismatched too.

Strangely only half of the stickers remained unpainted after painting them as you can see in the image I posted, the whole edge of the wing uses grey stickers but half of them show up as white on the preview
